zyte_spider_templates/spiders/serp.py

```diff
@@ -0,0 +1,131 @@
+from typing import Any, Dict, Iterable, List, Optional, Union
+
+from pydantic import BaseModel, Field, field_validator
+from scrapy import Request
+from scrapy.settings import SETTINGS_PRIORITIES, BaseSettings
+from scrapy_spider_metadata import Args
+from w3lib.url import add_or_replace_parameter
+from zyte_common_items import Serp
+
+from ..params import MaxRequestsParam
+from ._google_domains import GoogleDomain
+from .base import BaseSpider
+
+
+class SearchQueriesParam(BaseModel):
+    search_queries: Optional[List[str]] = Field(
+        title="Search Queries",
+        description=(
+            "Input 1 search query per line. A search query is a string of "
+            "search keywords (e.g. foo bar)."
```

Comment:
@Gallaecio as we discussed on a call, probably better to avoid mentioning keywords